### PR TITLE
[System] Common NetworkInformation code from referencesource

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: mcs/class/referencesource/System/net/System/Net/NetworkInformation/SystemUnicastIPAddressInformation.cs [hunk 1]
```diff
@@ -24,6 +24,9 @@ internal class SystemUnicastIPAddressInformation:UnicastIPAddressInformation {
         private uint preferredLifetime;
         private byte prefixLength;
 
+#if MONO
+
```
Comment (akoeplinger) [inline mcs/class/referencesource/System/net/System/Net/NetworkInformation/SystemUnicastIPAddressInformation.cs]: why is this empty?

File: mcs/class/System/System.Net.NetworkInformation/NetworkInformationException.cs [hunk 2]
```diff
@@ -1,74 +0,0 @@
...
-using System.Runtime.Serialization;
-
-namespace System.Net.NetworkInformation {
-	[Serializable]
-	public class NetworkInformationException
-#if !NET_2_1
-		: Win32Exception
-#else
-		: Exception
-#endif
```
Comment (akoeplinger) [inline mcs/class/System/System.Net.NetworkInformation/NetworkInformationException.cs]: The RS version of this class doesn't have the `NET_2_1` changes, is this intentional?

File: mcs/class/System/System.Net.NetworkInformation/GatewayIPAddressInformationCollection.cs [hunk 3]
```diff
@@ -1,156 +0,0 @@
...
-			get { return true; }
-		}
-
-		public virtual GatewayIPAddressInformation this [int index] {
-			get { return list [index]; }
-		}
-	}
-
-#if !MOBILE
-	class Win32GatewayIPAddressInformationCollection : GatewayIPAddressInformationCollection
```
Comment (akoeplinger) [inline mcs/class/System/System.Net.NetworkInformation/GatewayIPAddressInformationCollection.cs]: This class was excluded from the mobile build in https://github.com/mono/mono/commit/9ea2f50a2adefdeadefbf8f49aa94404e3175a1f, but in the new code it isn't. Do the reasons from that commit still apply?

File: mcs/class/referencesource/System/net/System/Net/NetworkInformation/SystemUnicastIPAddressInformation.cs (hunk 1 above)
Comment (marek-safar) [inline mcs/class/referencesource/System/net/System/Net/NetworkInformation/SystemUnicastIPAddressInformation.cs]: That's due to future work but will kill it for now

File: mcs/class/System/System.Net.NetworkInformation/NetworkInformationException.cs (hunk 2 above)
Comment (marek-safar) [inline mcs/class/System/System.Net.NetworkInformation/NetworkInformationException.cs]: Yes, that's intentional. I don't see any reason why customize it and then hunt RS code for wrong catches

Comment (akoeplinger) [inline mcs/class/System/System.Net.NetworkInformation/NetworkInformationException.cs]: I assumed there was a reason why we don't want this class to inherit from Win32Exception on mobile. If that's not the case, I'm of course fine with the cleanup.

Comment (marek-safar) [inline mcs/class/System/System.Net.NetworkInformation/NetworkInformationException.cs]: The reason was Silverlight

**Pull request merged**